File: common/src/main/java/com/rabtman/common/integration/IDbMigrationManager.java
package com.rabtman.common.integration;

import io.realm.RealmMigration;

/**
 * @author Rabtman
 */

public interface IDbMigrationManager extends RealmMigration {

  /**
   * 使用{@link RealmMigration}注入各个组件中编写的迁移
   */
  void injectDbMigrations(RealmMigration... dbMigrations);
}

File: common/src/main/java/com/rabtman/common/integration/IDbMigrationManager.kt
package com.rabtman.common.integration

import io.realm.RealmMigration

/**
 * @author Rabtman
 */
interface IDbMigrationManager : RealmMigration {
    /**
     * 使用[RealmMigration]注入各个组件中编写的迁移
     */
    fun injectDbMigrations(vararg dbMigrations: RealmMigration)
}